File: FrontEnd/signup1/lib/Tutor/course.dart
class Courses{
  final String ?course_name,topic_id,course_duration,course_price,tutor_id,course_image;

  Courses(this.course_name, this.topic_id, this.course_duration, this.course_price,this.tutor_id,this.course_image);
  Courses.fromJson(Map<String,dynamic>json, this.course_name, this.topic_id, this.course_duration, this.course_price,this.tutor_id,this.course_image);
  Map<String,dynamic> toJson()=>{
    "course_name":this.course_name,
    "topic_id":this.topic_id,
    "course_duration":this.course_duration,
    "course_price":this.course_price,
    "tutor_id":this.tutor_id,
    "course_image":this.course_image
  };
}

File: BackEnd/node js/model/course.js
var mongoose = require("mongoose")
const {ObjectId}=require('mongodb')
var courseSchema=mongoose.Schema({
    course_name:{
        type: String,
        required: true,
        minlength: 3,
        maxlength: 50,
    },
    topic_id:{
        type: ObjectId,
        required: true,
    },
    course_duration:{
        type:String,
        required:true
    },
    course_price:{
        type:String,
        required:true
    },
    tutor_id:{
        type:ObjectId
    },
    course_image:{
        type:String
    },
    course_status:{
        type:Number,
        default:0
    },
})

module.exports=mongoose.model("Course",courseSchema)